app.py

```diff
@@ -169,15 +169,21 @@ def before_request_https():
 
 # Unique random key for sessions.
 # For settings with multiple workers, an environment variable is required, otherwise cookies will be constantly removed and re-set by different workers.
-app.config['SECRET_KEY'] = os.getenv ('SECRET_KEY') or uuid.uuid4().hex
+if utils.is_production():
+    if not os.getenv ('SECRET_KEY'):
+        raise RuntimeError('The SECRET KEY must be provided for non-dev environments.')
+
+    app.config['SECRET_KEY'] = os.getenv ('SECRET_KEY')
 
-# Set security attributes for cookies in a central place - but not when running locally, so that session cookies work well without HTTPS
-if os.getenv ('HEROKU_APP_NAME'):
     app.config.update(
```

Comment:
I think it'd be prudent to have the same cookie attributes in `alpha` and `test`, if only to make sure that there's no unexpected behaviors that can only be tested in `beta`.